src/main/java/com/gu/email/exacttarget/ExactTargetFactory.java

```diff
@@ -2,97 +2,78 @@
 
 import com.gu.email.AccountDetails;
 import com.gu.email.GuardianUser;
-import org.apache.commons.httpclient.methods.PostMethod;
-import org.apache.commons.httpclient.methods.RequestEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.fluent.Request;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.util.EntityUtils;
 import org.jdom.Document;
 import org.jdom.JDOMException;
 import org.jdom.input.SAXBuilder;
 
 import java.io.ByteArrayInputStream;
 import java.io.IOException;
-import java.io.InputStream;
 import java.net.URI;
-import java.util.HashMap;
 import java.util.Map;
 
 
-public class ExactTargetFactory
-{
+public class ExactTargetFactory {
     public final URI endPoint;
     private final AccountDetails accountDetails;
 
-    public ExactTargetFactory( String accountName, String password, URI endPoint )
-    {
+    public ExactTargetFactory(String accountName, String password, URI endPoint) {
         accountDetails = new AccountDetails(accountName, password);
         this.endPoint = endPoint;
     }
 
-    public PostMethod createPostMethod( RequestEntity body, String soapAction )
-    {
-        PostMethod method = new PostMethod( endPoint.toString() );
-
-        method.setRequestHeader( "Host", endPoint.getHost() );
-        method.setRequestHeader( "Content-Type", "text/xml; charset=utf-8" );
-        method.setRequestHeader( "Content-Length", "" + body.getContentLength() );
-        method.setRequestHeader( "SOAPAction", soapAction );
-        method.setRequestEntity(body);
-
-        return method;
+    public Request createPostMethod(StringEntity body, String soapAction) {
+        return Request
+                .Post(endPoint)
```

Comment:
content type is set automatically?

Reply:
yeah it's now part of `StringEntity` and set automatically.